ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue

```diff
@@ -40,91 +40,138 @@
+                <template v-slot:activator="{ on }">
+                  <v-btn text v-on="on"
+                         color="primary" class="px-0"
+                         :disabled="isAddingMode || isGlobalEditingMode"
+                  >
+                    <v-icon>mdi-menu-down</v-icon>
+                  </v-btn>
+                </template>
 
-              <!-- More actions drop down list -->
-              <v-list class="actions-dropdown actions__more-actions">
-                <v-list-item class="my-n2">
-                  <v-list-item-subtitle class="pa-0" @click="remove(row.item)">
-                    <v-icon small style="margin-bottom: 3px;">mdi-delete</v-icon>
-                    <span class="ml-1 remove-btn-text">Remove</span>
-                  </v-list-item-subtitle>
-                </v-list-item>
-              </v-list>
-            </v-menu>
+                <!-- More actions drop down list -->
+                <v-list class="actions-dropdown actions__more-actions">
+                  <v-list-item class="my-n2">
+                    <v-list-item-subtitle class="pa-0" @click="remove(row.item)">
+                      <v-icon small style="margin-bottom: 3px;">mdi-delete</v-icon>
+                      <span class="ml-1 remove-btn-text">Remove</span>
+                    </v-list-item-subtitle>
+                  </v-list-item>
+                </v-list>
+              </v-menu>
+            </template>
+
+            <!-- Existing Owner Actions -->
+            <template v-else>
+              <v-btn
+                v-if="!isRemovedHomeOwner(row.item)"
+                text color="primary" class="pr-0"
+                :ripple="false"
+                :disabled="isAddingMode || isEditingMode || isGlobalEditingMode"
+                @click="markForRemoval(row.item)"
+                data-test-id="table-edit-btn"
+              >
+                <v-icon small>mdi-delete</v-icon>
+                <span>Delete</span>
+              </v-btn>
+              <v-btn
+                v-if="isRemovedHomeOwner(row.item)"
+                text color="primary" class="pr-0"
+                :ripple="false"
+                :disabled="isAddingMode || isEditingMode || isGlobalEditingMode"
+                @click="undoRemoval(row.item)"
+                data-test-id="table-edit-btn"
+              >
+                <v-icon small>mdi-undo</v-icon>
+                <span>Undo</span>
+              </v-btn>
+            </template>
           </td>
         </tr>
         <tr v-else>
-          <td :colspan="4" class="py-1">
+          <td v-if="showGroups" :colspan="4" class="py-1">
             <div
-              v-if="showGroups"
               class="error-text my-6 text-center"
               :data-test-id="`no-owners-msg-group-${homeOwners.indexOf(row.item)}`"
             >
               Group must contain at least one owner
             </div>
-            <div v-else class="my-6 text-center" data-test-id="no-owners-mgs">
```

Comment:
What's the reason for this removal?

Reply:
There was a bug here. This cell would always render, so after removing owner it would default to 'no owners added yet' msg, in the event we were not dealing with groups. 
Down below there is messaging already for that no?

Reply:
Those are different scenarios for Registrations (from what I remember). One would render initially when there is no data, and the other one would render after owners were removed.

Reply:
I tested this removal in Registrations and i couldn't find a scenario where there was no msg displayed in the absence of an Owner.

Reply:
If you add a solo owner and then remove it - this condition is triggered and a message shows up.

Reply:
Ah fair enough, it's a bug with the data being cleared when the owner is removed. This additional conditional template was just masking it.

<img width="961" alt="Screen Shot 2022-10-18 at 3 23 01 PM" src="https://user-images.githubusercontent.com/53542131/196556543-97eb8c2b-268f-4a9c-92bd-11754c82eca5.png">

Reply:
It should resolve to no data but the groupId is persisting after clearing the new added owner in Registrations. 
Can fix.

Reply:
Yea, not ideal, but there was a reason for this. We need to preserve the Groups with no Owners, I think this was the fix for it.

Reply:
Hmm ok, will take another look. 
This work around was causing issues with Transfers.

Reply:
All refactored, removed individual arguments and leaning on Composable argument.

Reply:
Just a note, we may need to revisit this as we explore the removal of Groups in Transfer Flow. 
Coming soon *TM